src/kak-setsid.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+#include <unistd.h>
+
+void execute_command_in_arguments(int argc, char *argv[]);
+
+int main(int argc, char *argv[])
+{
+    if(argc < 2) {
+        fprintf(stderr, "Give a command to execute as argument to %s.\n",
+                argv[0]);
+        return 1;
+    }
+    
+    switch(fork()) {
+        case -1:
+            fprintf(stderr, "Failed to create child process.\n");
+            return 2;
+        case 0:
+            setsid();
+            execute_command_in_arguments(argc, argv);
+            return 3;  /* execute_command_in_arguments() should not return. */
+        default:
+            break;
```

Comment:
So the second and third things I've learned about C99 today is that you don't need a return statement in main() (it will return 0) and that you can rely on return statements in main() instead of needing to call exit()!  I have apparently been out of touch.

However, I think it's still a good idea to `return 0;` here to be explicit.

Reply:
Done.